app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -7,12 +7,18 @@ window.GOVUK.analyticsGA4.analyticsModules = window.GOVUK.analyticsGA4.analytics
   'use strict'
 
   var Ga4LinkTracker = {
-    init: function () {
+    init: function (config = {}) {
       if (window.dataLayer) {
-        this.internalLinksDomain = 'www.gov.uk/'
-        this.internalLinksDomainWithoutWww = 'gov.uk/'
+        this.config = config
```

Comment:
I'm not clear on where this config will be passed from once the code is in use. Can you provide more detail?

Reply:
@andysellick Good point - I think initially I thought I'd have to surround `stopTracking()` with a check for `!this.config.disableListeners`, but it seems to still work without this check, so I had accidentally left it as `this.config`

Reply:
I've changed it to just be `config` now.

Reply:
I seem to have stumbled upon something else accidentally! Good spot.

What I meant was that this whole block of code receives `config` as an optional parameter, but what calls this code and passes that?

Reply:
@andysellick The tests call this so that it can test against `www.gov.uk` while the current hostname is `localhost`. It was also helpful in case we wanted to initialise the link tracking without the click listeners. It also allows for extra internal domains / download domains / download paths to be added in case there are multiple domains to class as internal, or multiple domains to be classed as download links, or multiple paths to be classed as download paths. It essentially removes the hard coding of the functionality only working for `www.gov.uk`'s MVP purposes and allows it to be a bit more future proof.

Reply:
That sounds good. For future reference for testing, you can use a `spyOn` to override a function like `getHostname` so that it returns whatever you want, something like:

```
spyOn(Ga4LinkTracker, 'getHostname').and.returnValue('www.whateveryouwant.com')
```

Reply:
Thanks @andysellick - I think at one point when I initially built this I was trying to spy on `window.location.hostname` which it wouldn't let me do, I didn't think of just wrapping it in a function!